### PR TITLE
Fix Forte [Psi Part]

### PR DESCRIPTION
**Pull request by JonathonMisiewicz**
## Description
This PR makes changes so that `Psi` can do DIIS on an `ambit.BlockedTensor`, as required by the `forte` plugin. This PR will not work until [this ambit PR](https://github.com/jturney/ambit/pull/53) is merged, but passes locally. The test suite tells us little about this one, sadly.

@loriab, I leave it up to you whether this warrants an update to the version of ambit that Psi takes.

@fevangelista @lcyyork

## Status
- [x] Ready for review
- [ ] Leave merging to me

### CODE REVIEW

**Review by loriab — APPROVED** [review]
I don't think psi fails in the presence of the longstanding ambit with this PR, so no need to insist on new tag. You could update https://github.com/psi4/psi4/blob/master/external/upstream/ambit/CMakeLists.txt#L30 here once the ambit PR is merged (do a few characters of sha instead of the tag).

lgtm!

File: psi4/driver/procrouting/diis.py [hunk 1]
```diff
@@ -134,12 +167,17 @@ def load_quantity(self, name, entry_num, item_num, force_new = True):
...
+            if hasattr(entry_dims, "__len__"):
+                if len(entry_dims) == 2:
+                    quantity = core.Matrix(full_name, *entry_dims)
+                    quantity.load(psio, psif.PSIF_LIBDIIS, core.SaveType.SubBlocks)
+                elif len(entry_dims) == 1:
+                    quantity = core.Vector(full_name, *entry_dims)
+                    quantity.load(psio, psif.PSIF_LIBDIIS)
+            elif which_import("ambit", return_bool=True):
+                import ambit
+                if entry_dims == ambit.BlockedTensor:
```
[the remark this answers is not in the document]
Comment (JonathonMisiewicz) [reply psi4/driver/procrouting/diis.py:179]: Not `isinstance` here. The dimension information is written to disk along with the actual files, so in this case, all I need to know is that we have an `ambit.BlockedTensor`, not its dimensions. For that reason, I store [the `BlockedTensor` class itself](https://github.com/psi4/psi4/pull/2505/files#diff-0106e9981286de9bf2e7872ff90480bd726d056e61a5a39baef8739714fa148fR52). This was what I was getting at [in these new comments](https://github.com/psi4/psi4/pull/2505/files#diff-0106e9981286de9bf2e7872ff90480bd726d056e61a5a39baef8739714fa148fR34-R40).

I'm happy to change variable names or add more comments if you can find a way to make things clearer. Perhaps rename `entry_dims`, now that it's no longer necessarily dimension information?

**Review by jturney — APPROVED** [review]

File: psi4/driver/procrouting/diis.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (loriab) [reply psi4/driver/procrouting/diis.py:179]: ok, thanks. fine as-is, then. could you skip the `entry_dims = template_object` step? the latter name is a little clearer.

**Review by fevangelista — APPROVED** [review]

**Comment by zachglick** [comment]
This is a nice feature! Is there a good place (perhaps a docstring) to explicitly state that DIIS is now compatible with ambit tensors?

**Comment by JonathonMisiewicz** [comment]
> This is a nice feature! Is there a good place (perhaps a docstring) to explicitly state that DIIS is now compatible with ambit tensors?

Added.

Some changes were requested in the `ambit` PR, so I'll need to double-check that the `forte` tests still work before I can merge this in.

**Pull request merged**